packages/core-mobile/app/new/features/deposit/context/DepositContext.tsx

```diff
@@ -0,0 +1,118 @@
+import { Network } from '@avalabs/core-chains-sdk'
+import { TokenUnit } from '@avalabs/core-utils-sdk'
+import React, {
+  createContext,
+  Dispatch,
+  ReactNode,
+  useContext,
+  useMemo,
+  useState
+} from 'react'
+import useCChainNetwork from 'hooks/earn/useCChainNetwork'
+import { getViemChain } from 'utils/getViemChain/getViemChain'
+import { createPublicClient, http } from 'viem'
+import { LocalTokenWithBalance } from 'store/balance'
+import { useTokensWithBalanceByNetworkForAccount } from 'features/portfolio/hooks/useTokensWithBalanceByNetworkForAccount'
+import { selectActiveAccount } from 'store/account'
+import { useSelector } from 'react-redux'
+import { DefiAssetDetails, DefiMarket } from '../types'
+import { useAvailableMarkets } from '../hooks/useAvailableMarkets'
+
+interface DepositContextState {
+  amount: TokenUnit | undefined
+  setAmount: Dispatch<TokenUnit | undefined>
+  network: Network | undefined
+  resetAmount: () => void
+  markets: DefiMarket[]
+  isLoadingMarkets: boolean
+  depositableTokens: DefiAssetDetails[]
+  tokensWithBalance: LocalTokenWithBalance[]
+}
+
```

Comment:
I would use zustand completely for this. one downside of using context is it will re-render all its children when some state changes. I assume isLoadingMarkets and tokensWithBalance will change very frequently. another thing is tokensWithBalance is already stored and managed by theuseTokensWithBalanceByNetworkForAccount hook. there is no need to store it again here in context or zustand.